lib/_stream_writable.js

```diff
@@ -784,6 +782,20 @@ function onFinished(stream, state, cb) {
   stream.prependListener('error', onerror);
 }
 
+ObjectDefineProperty(Writable.prototype, 'writable', {
+  get() {
+    const w = this._writableState;
+    if (w && w.writable !== undefined) return w.writable;
+    return Boolean(w && !w.destroyed && !w.errored && !w.ending);
+  },
+  set(val) {
+    // Backwards compat.
+    if (this._writableState) {
+      this._writableState.writable = val;
```

Comment:
I would cast these to booleans

Reply:
Doing so breaks some tests :(

Reply:
which ones?

Reply:
http2 has some tests where it checks for `stream.readable=0; assert.strictEqual(stream.readable, 0)`

Reply:
```

AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

false !== 0

    at Http2Server.<anonymous> (/Users/ronagy/GitHub/nxtedition/node/test/parallel/test-http2-socket-proxy.js:90:10)
```

Reply:
Would you prefer the cast + change http2 test?

Reply:
Given that this is semver-major, I would go ahead and cast to boolean and update the http2 impl to match.